src/rtk/features/posts/postsSlice.ts

```diff
@@ -427,6 +418,159 @@ export const fetchPosts = createAsyncThunk<PostStruct[], FetchPostsArgs, ThunkAp
 
 export const fetchPost = createFetchOne(fetchPosts)
 
+type FetchProfilePostsArgs = {
+  id: AccountId
+  spaceId?: string
+  withHidden?: boolean
+  dispatch: AppDispatch
+  api: SubsocialApi
+  client?: ApolloClient<NormalizedCacheObject> | undefined
+  limit?: number
+  offset?: number
+
+  additionalData?: {
+    withContent?: boolean
+    withSpace?: boolean
+    withOwner?: boolean
+    withRootPost?: boolean
+  }
+}
+
+export const fetchProfilePosts = createAsyncThunk<
+  PostStruct[],
+  FetchProfilePostsArgs,
+  ThunkApiConfig
+>(
+  'posts/fetchOneByAddress',
+  async ({
+    id,
+    spaceId,
+    client,
+    additionalData,
+    withHidden,
+    dispatch,
+    api,
+    limit,
+    offset,
+  }): Promise<PostStruct[]> => {
+    const address = id as AccountId
+
+    if (!client) return []
+
+    const showHidden = withHidden ? {} : { hidden_eq: false }
+
+    const posts = await getPostsData(client, {
+      offset,
+      limit,
+      where: {
+        ownedByAccount: { id_eq: address },
+        AND: [{ space_isNull: false, isComment_eq: false, ...showHidden }],
+        OR: [{ space: { id_eq: spaceId }, ...showHidden }],
+      },
+    })
+
+    const {
+      withContent = true,
+      withSpace = true,
+      withOwner = true,
+      withRootPost = true,
+    } = additionalData || {}
+
+    const fixedPosts = await Promise.all(
+      posts.map(async post => {
+        if (!post.contentId && post.ipfsContent) {
+          const newCid = xxhashAsHex(Buffer.from(JSON.stringify(post.ipfsContent)), 128, true)
+          post.contentId = Buffer.from(newCid).toString('hex')
+        }
+        return post
+      }),
+    )
+
+    const entities = fixedPosts.map<PostState>(post => ({ ...post, isOverview: true }))
+
+    const generatePrefetchData = generatePrefetchDataFn<PostState, PostFragmentWithParent>(entities)
+
+    const fetches: Promise<any>[] = []
+    if (withOwner) {
+      const ids = getUniqueOwnerIds(entities)
+      const prefetchedData = generatePrefetchData<ProfileSpaceIdByAccount>(
```

Comment:
Check which part of code you use in both fetch functions, and reuse it instead of copy-pasting